src/common/filters/not-found-exception.filter.ts

```diff
@@ -5,9 +5,10 @@ import {
 	HttpException,
 	NotFoundException,
 } from '@nestjs/common';
+import { Prisma } from '@prisma/client';
 import { Response } from 'express';
 
-@Catch(NotFoundException)
+@Catch(NotFoundException, Prisma.PrismaClientKnownRequestError)
```

Comment:
This will catch all known Prisma client errors, even when a conflict arises.

We should check the code error to validate that the error is indeed a 404 error (with a quick look, errors `P2001`, `P2015`, `P2018` or `P2025`): <https://www.prisma.io/docs/orm/reference/error-reference#error-codes>.